media-session/index.js

```diff
@@ -0,0 +1,119 @@
+let audio = document.createElement('audio');
+
+let playlist = getPlaylist();
+let index = 0;
+
+function onPlayButtonClick(event) {
+  playAudio();
+}
+
+function playAudio() {
+  audio.src = playlist[index].src;
+  audio.play()
+  .then(_ => updateMetadata())
+  .catch(error => log(error));
+}
+
+function updateMetadata() {
+  let track = playlist[index];
+
+  log('Playing "' + track.title + '" track...');
+  navigator.mediaSession.metadata = new MediaMetadata({
+    title: track.title,
+    artist: track.artist,
+    album: track.album,
+    artwork: track.artwork
+  });
+}
+
+/* Previous Track & Next Track */
+
+navigator.mediaSession.setActionHandler('previoustrack', function() {
+  log('> User clicked "Previous Track" icon.');
+  index = (index - 1 + playlist.length) % playlist.length;
+  playAudio();
+});
+
+navigator.mediaSession.setActionHandler('nexttrack', function() {
+  log('> User clicked "Next Track" icon.');
+  index = (index + 1) % playlist.length;
+  playAudio();
+});
+
+/* Seek Backward & Seek Forward */
+
+let skipTime= 10; /* Time to skip in seconds */
+
+navigator.mediaSession.setActionHandler('seekbackward', function() {
+  log('> User clicked "Seek Backward" icon.');
+  audio.currentTime = Math.max(audio.currentTime - skipTime, 0);
+});
+
+navigator.mediaSession.setActionHandler('seekforward', function() {
+  log('> User clicked "Seek Forward" icon.');
+  audio.currentTime = Math.min(audio.currentTime + skipTime, audio.duration);
+});
+
+/* Utils */
+
+function getPlaylist() {
+  return [{
+      src: 'jam.ogg',
```

Comment:
@samdutton That is my main issue there. Instead of having to upload one single audio file, I'd like to use some CC audio files from our server. Do you know if we already have some?